### ISSUE TITLE
How to require value for option if provided, but not require option itself?

### ISSUE THREAD

**Issue by platinumazure** [gkz/optionator#19]
I'm trying to create an option which requires a string argument if the option is used, but the option itself isn't required. I need to be able to distinguish between (option wasn't supplied) vs (option was supplied, but with no/invalid argument). Unfortunately, right now it seems both scenarios coalesce to the same option value for all of the type specifications I've tried.

Type specifications I've tried:
* String
* Maybe String
* Null | String
  * This was with the hope that undefined would be a default value if the option was specified and had no value

Maybe I'm missing something? But in case I'm not, I'll offer a couple of potential solutions that could work for me. (I'm completely open to other ideas, I'm just trying to show what sorts of things could help that I'm aware of.)

## New "default"-like option specifically for one of the two cases

If there was, for example, a "defaultIfNotUsed" option-level option, then optionator could use that value if the option is never specified and the "default" value if the option is specified with no value.

## Sentinel value that could be set as default

I'm open to other options, though. I could also see (for example) a sentinel `optionator.unspecified` (set to some `{}` object and used for reference checks), and a corresponding option-level option indicating that the unspecified object should be used if the option is not provided at all.

**Comment by platinumazure** [comment]
Actually, you know what might be simpler-- just a `requiresValueIfSpecified: true` option-level option, which would be orthogonal to option value validation.

Like I said earlier, I'm open to all sorts of solutions on this.

**Comment by gkz** [comment]
Can you give an example use-case and why it didn't work with the type "String"?

**Comment by platinumazure** [comment]
I'm trying to reconfigure eslint's `--print-config` option so that it actually requires an argument (right now it's a boolean option and we check for 1 positional file argument, which feels kind of weird).

Option should require a value if provided, but if the option is not provided, then there should be no validation error.

Valid:

```
eslint --print-config someFile.js

eslint --some-other-option
```

Invalid:

```
eslint --print-config
```

What ended up happening when I used String or Maybe String is, the `eslint --print-config` scenario and the `eslint --some-other-option` (i.e., don't include --print-config) both resulted in the value being undefined or null, depending on the default I set up. There was no way to distinguish between the two scenarios at all.

Does that help?

**Comment by gkz** [comment]
I think this is the same bug as #18

**Comment by gkz** [comment]
Ie. If the type of your option is anything other than Boolean, specifying a value should always be required. Currently there is a bug where this check is not happening if the flag is the last thing specified.

**Comment by gkz** [comment]
I have published 0.8.2, let me know if it solves your problem.

**Comment by platinumazure** [comment]
I didn't think of trying a test case `eslint --print-config --some-other-option`, sorry. I can try that tonight and see if this is fixed with 0.8.2.
